File: services/teams-service/Models/TeamLink.cs
using System.Text.Json.Serialization;

namespace TeamsService.Models
{
    public class TeamLink : BaseModel
    {
        public string Name { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public string? Text { get; set; }
        public string? IconType { get; set; }
        public bool IsPrivate { get; set; } = false;
        public int TeamId { get; set; }

        [JsonIgnore]
        public Team? Team { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: services/main-service/app/Models/TeamLink.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class TeamLink extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'url',
        'text',
        'icon_type',
        'is_private',
        'team_id'
    ];

    /**
     * @return BelongsTo
     */
    public function team(): BelongsTo
    {
        return $this->belongsTo(Team::class);
    }
}
